bin/pre-commit

```diff
@@ -24,14 +24,17 @@ if [[ -f WORKSPACE ]]; then
   ROOT=$PWD
 fi
 
-cd $ROOT
+# If we don't have a branch then we're in a detached state (e.g. mid rebase) and don't want to execute our checks.
+BRANCH_NAME=$(git branch | grep '*' | sed 's/* //')
+if [ $BRANCH_NAME != '(no branch)' ]; then
+    cd $ROOT
 
-# While adding more tests here keep in mind that
-# this runs every time a user commits
-# Run all the formatters now
-bin/fmt.sh
+    # While adding more tests here keep in mind that
+    # this runs every time a user commits
+    # Run all the formatters now
+    bin/fmt.sh
```

Comment:
now that you are touching this file :-- 
we should *only* run bin/fmt.sh  on pre-commit

bazel test //...
bin/linters.sh
bin/racetest.sh 

should all be put in the  pre-push hook.